controllers/managedClusterResources.go

```diff
@@ -592,13 +588,15 @@ func (r *ClusterGroupUpgradeReconciler) deployWorkload(
 		spec.ViewUpdateIntervalSec = utils.ViewUpdateSec * len(clusterGroupUpgrade.Status.Precaching.Clusters)
 		r.Log.Info("[deployPrecachingWorkload]", "getPrecacheJobTemplateData",
 			cluster, "status", "success")
+
 	case backup:
-		spec = r.getBackupJobTemplateData(clusterGroupUpgrade, cluster)
+		spec, err = r.getBackupJobTemplateData(clusterGroupUpgrade, cluster)
```

Comment:
same as above in the other comment.

Reply:
err is initialized before the switch, thats why probably. Overlooked previously